### PR TITLE
smfc: go generator fixes

### LINKED ISSUE #286: SMF Go generator assumes flatbuffer names and directory structures
**Describe the bug**

The go code generator within smfc is a nice feature, as it allows for some quick prototyping and use of non-Linux platforms for testing.

That being said, the code generator has some hardcoded paths in it that cause `go build` to reject the output code.  Example failed `go build` output includes:

```
# not_a_demo
../not_a_demo/not_demo_service.smf.fb.go:13:24: undefined: smf.Request
../not_a_demo/not_demo_service.smf.fb.go:37:32: undefined: demo
../not_a_demo/not_demo_service.smf.fb.go:48:68: undefined: not_a_demo
../not_a_demo/not_demo_service.smf.fb.go:53:9: undefined: smf.GetRootAsResponse
```

The first two and the last one appear to be the result of hardcoding for specific expected types that don't always exist (`Request`, `Response`, `Demo`). 

The third one uses the same package as the current flatbuffer protocol file to identify a the response flatbuffer type, which is fine if the generated service is in a different package.  In my example, below, this is not the case.

**To Reproduce**
1. Set `GOPATH` to something appropriate, and `mkdir -p $GOPATH/src`
2. `cd $GOPATH/src`, and get dependencies:

```
cd $GOPATH/src
go get github.com/google/flatbuffers/go
go get github.com/smfrpc/smf-go/src/smf
```

3. Create the following flatbuffer file (`not_a_demo.fbs`):

```
namespace not_a_demo;

table Request1 {
  name: string;
}

table Response1 {
  name: string;
}

rpc_service SmfStorage1 {
  Get(Request1):Response1;
}
```

3. Invoke `flatc` and `smfc` on it (ensure you're in the `${GOPATH}/src` directory):

```
/path/to/flatc --gen-name-strings --gen-object-api --go --force-empty --gen-compare --keep-prefix --json --reflect-names --defaults-json --gen-mutable -o . /path/to/not_a_demo.fbs 

/path/to/smfc --filename not_demo_service.fbs --language=go --output_path=.
```

4. Make a directory for the "main" module:

```
mkdir not_a_demo_client
```

5. Create a `not_a_demo_client/main.go` file containing the following:

```go
package main

import (
        "fmt"
        "not_a_demo"
)

func main() {
        fmt.Printf("It compiled!")
}
```

6. `cd` to `not_a_demo_client` and attempt a build:

```
cd not_a_demo_client
go build
```

The compilation fails as above.

**Expected behavior**
Output code to contain correct descriptions of types, and thusly, for compilation to succeed.

** Environment Info **

- OS: Ubuntu 16.04
- smf version: git-9dfa658
- smf-go version: git-91025c4
- Compiler: g++-7
- Go Version: 1.11.2

### PR DESCRIPTION
**Pull request by crackcomm**
Fixes #286

### CODE REVIEW

File: src/smfc/go_generator.h [hunk 1]
```diff
@@ -41,10 +41,6 @@ class go_generator : public generator {
     generate_header_includes();
     generate_header_services();
 
-    // for Go make sure that the directories exist
```
Comment (crackcomm) [inline src/smfc/go_generator.h]: so many hours

Comment (senior7515) [reply src/smfc/go_generator.h]: thanks!

File: src/smfc/go_generator.cc [hunk 2]
```diff
@@ -25,14 +26,15 @@ print_server_method(smf_printer &printer, const smf_method *method) {
   vars["MethodID"] = std::to_string(method->method_id());
   vars["RawMethodName"] = "Raw" + go_public_name(method->name());
   vars["MethodName"] = go_public_name(method->name());
+  vars["InType"] = method->input_type_name(language::go);
```
Comment (senior7515) [inline src/smfc/go_generator.cc]: nice fix

**Comment by senior7515** [comment]
@crackcomm   next time, please sign off your commits via 
```
git commit -s ..............
```

File: src/smfc/go_generator.h (hunk 1 above)
Comment (senior7515) [reply src/smfc/go_generator.h]: @crackcomm  - I think this hunk is a regression. 

if you have an flatbuffers package of `foo.bar.baz` it should create `/foo/bar/baz/*_generated.go`

Can you comment on go experience here. I think i need to revert this  directory creation

Comment (senior7515) [reply src/smfc/go_generator.h]: @crackcomm  - so i fixed the bug upstream for consistency on https://github.com/smfrpc/smf/commit/d20866d5b16ffa251bd0b0bbff4d777e8480df7c

however the previous behavior matches that of flatc - so my incline is to keep the old behavior. 

Thoughts?

Comment (crackcomm) [reply src/smfc/go_generator.h]: It was creating an empty directory with files outside of it...

**Pull request merged**